orderbook/orderbook.c

```diff
@@ -7,6 +7,12 @@ associated with this software.
 #include "orderbook.h"
 #include "utils.h"
 
+typedef int (*string_builder_t)(PyObject *pydata, uint8_t *data, int *pos);
+
+// reusable python objects referring to the builtin format function and a string 'f'
+static PyObject* format = NULL;
+static PyObject* formatf = NULL;
```

Comment:
dont format and formatf need to be cleaned up to avoid a memory leak?

Reply:
Strictly speaking yes, but only if one manages to de-allocate the module (by default they are kept in cache even when executing `del module`) and re-import it. So this use-case is uncommon but not impossible.

I agree though that it would be neater to clean them and ideally also to avoid these global variables by using [PyModule_GetState()](https://python.readthedocs.io/fr/latest/c-api/module.html#c.PyModule_GetState). I suspect the best starting point would be https://peps.python.org/pep-0489/, but if you have other thoughts I'd be happy to hear them.

Reply:
This issue is resolved in the latest commit.